test/error.ts

```diff
@@ -359,6 +361,21 @@ test.skip('the old stacktrace is recovered', async t => {
 	t.not(error?.stack!.indexOf('at get'), error?.stack!.lastIndexOf('at get'));
 });
 
+test('should wrap got cause', async t => {
+	const error = await t.throwsAsync<RequestError>(got('https://github.com', {retry: {limit: 0}, timeout: {request: 1}}));
+	const cause = error?.cause as TimeoutError;
+	t.is(error?.code, cause.code);
+	t.is(error?.message, cause.message);
+});
+
+test('should wrap non-got cause', async t => {
+	const error = await t.throwsAsync<RequestError>(got('https://github.com', {retry: {limit: 0}, timeout: {read: 1}, agent: {https: new SocksProxyAgent('socks://your-name%40gmail.com:abcdef12345124@br41.nordvpn.com')}}));
```

Comment:
Can you do this test without adding two dev dependencies? Every added dependency adds a maintenance burden.